Realm/ObjectServerTests/RLMObjectServerTests.mm

```diff
@@ -1510,6 +1511,62 @@ - (void)testSetClientResetCallbacks {
 
 }
 
+- (void)testBeforeClientResetCallbackNotVersioned {
+    // Setup a realm with a versioned schema
+    RLMRealmConfiguration *config = [RLMRealmConfiguration defaultConfiguration];
+    config.fileURL = RLMTestRealmURL();
+    // Need to open the realm once to get a version.
+    @autoreleasepool {
+        RLMRealm *versioned = [RLMRealm realmWithConfiguration:config error:nil];
+        XCTAssertEqual(0U, versioned->_realm->schema_version());
+    }
+    std::shared_ptr<realm::Realm> versioned = realm::Realm::get_shared_realm(config.config);
+
+    // Create a config that's not versioned.
+    RLMRealmConfiguration *configUnversioned = [RLMRealmConfiguration defaultConfiguration];
+    configUnversioned.configRef.schema_version = RLMNotVersioned; // Not strictly necessary. Already has no version because the realm's never been opened.
```

Comment:
`schemaVersion` is set to 0 in [RLMRealmConfiguration's initializer](https://github.com/realm/realm-swift/blob/master/Realm/RLMRealmConfiguration.mm#L115). The schema version is not used for anything though because there's no schema set, so this test will pass with the schema version set to any value.

Reply:
Comments removed. Or do you think the whole lines could be removed since the schema isn't set?

Reply:
I think it's maybe misleading to have it, since it's not actually being used? Not really a big deal either way.